### PR TITLE
Rmkerr multiplatform upgrade

### PR DESCRIPTION
**Pull request by galanCA**
Upgrades rmkerr.cpp to work on all systems.

The changes added to this file are:
- Add iostream library
- Replace _MAX_PATH for FILENAME_MAX
- Replace unlink for c++ standard std::remove
- Replace getch for c standard. This was tested with a local unit test. It does require the user to press enter to mimic the preview effects.

## Author Progress Checklist:

- [ ] Open draft pull request
    - [ ] Make title clearly understandable in a standalone change log context
    - [ ] Assign yourself the issue
    - [ ] Add at least one label (enhancement, bug, or maintenance)
    - [ ] Link the issue(s) addressed by this PR (under "Development" in the sidebar menu) 
- [ ] Make code changes (if you haven't already)
- [ ] Self-review of code
    - [ ] My code follows the style guidelines of this project
    - [ ] I have added comments to my code, particularly in hard-to-understand areas
    - [ ] I have only committed the necessary changes for this fix or feature
    - [ ] I have made corresponding changes to the documentation
    - [ ] My changes generate no new warnings
    - [ ] I have ensured that my fix is effective or that my feature works as intended by:
        - [ ] exercising the code changes in the test framework, and/or
        - [ ] manually verifying the changes (as explained in the the pull request description above)
    - [ ] My changes pass all local tests
    - [ ] My changes successfully passes CI checks
    - [ ] Add any unit test for proof and documentation.
    - [ ] Merge in main branch and address resulting conflicts and/or test failures.
- [ ] Move pull request out of draft mode and assign reviewers
- [ ] Iterate with reviewers until all changes are approved
    - [ ] Make changes in response to reviewer comments
    - [ ] Merge in main branch and address resulting conflicts and/or test failures.
    - [ ] Re-request review in GitHub

## Reviewer Checklist:

 - [ ] Read the pull request description
 - [ ] Perform a code review on GitHub
 - [ ] Confirm all CI checks pass and there are no build warnings
 - [ ] Pull, build, and run automated tests locally
 - [ ] Perform manual tests of the fix or feature locally
 - [ ] Add any review comments, if applicable
 - [ ] Submit review in GitHub as either
     - [ ] Request changes, or
     - [ ] Approve
 - [ ] Iterate with author until all changes are approved

### CODE REVIEW

**Review by tanaya-bigladder — APPROVED** [review]

File: src/RMKERR.CPP [hunk 1]
```diff
@@ -21,8 +21,7 @@
 // OUTPNAMS: cond'ly def'd in cnglob.h to call "saveAnOutPNam" when error file is closed. 10-93.
 
 /*------------------------------- INCLUDES --------------------------------*/
-#include "cse.h"		// globals required for some configurations
-#include <conio.h>		// getch
+#include "CSE.h"		// globals required for some configurations
```
Comment (nealkruis) [inline src/RMKERR.CPP:24]: 🤔

File: src/RMKERR.CPP [hunk 2]
```diff
@@ -1028,7 +1029,8 @@ LOCAL int presskey( 	// prompt user after error message display
 		erAction == ABT
 		? "Press any key (program will abort) "
 		: "Press 'A' to abort, any other key to continue " );
-	int key = getch();							// input any char
+	int key;
+	std::cin >> key;							// input any char
```
Comment (nealkruis) [inline src/RMKERR.CPP:1033]: Did you test this? I'm pretty sure it is not equivalent behavior, though I don't know if there is a better alternative.

**Review by nealkruis — CHANGES_REQUESTED** [review]

File: src/rmkerr.cpp [hunk 3]
```diff
@@ -1026,9 +1027,10 @@ LOCAL int presskey( 	// prompt user after error message display
 #elif !defined( LOGWIN)
 	printf(
 		erAction == ABT
-		? "Press any key (program will abort) "
-		: "Press 'A' to abort, any other key to continue " );
-	int key = getch();							// input any char
+		? "Press any key and enter (program will abort) "
+		: "Press 'A' and enter to abort, any other key to continue " );
+	char key;
+	std::cin >> key;							// input any char
```
Comment (nealkruis) [inline src/rmkerr.cpp:1033]: This is still different behavior but, as I think you've concluded as well, I don't know if there is an alternative. I think the instructions could be a little clearer:

```
? "Enter any character (program will abort) "
: "Enter 'A' to abort, any other character to continue "
```

I'm also curious what happens if the user enters multiple characters.

**Review by nealkruis — CHANGES_REQUESTED** [review]

File: src/rmkerr.cpp (hunk 3 above)
Comment (galanCA) [reply src/rmkerr.cpp:1033]: As long as the first character is either `a` or `A` it will take that as an abort. Otherwise, it will continue.

**Review by nealkruis — APPROVED** [review]
This looks okay to me. @chipbarnaby are you okay with the slight change in behavior in the prompt?

**Review by tanaya-bigladder — APPROVED** [review]

**Pull request merged**